### PR TITLE
Animated status icons

### PR DESCRIPTION
**Pull request by Morb0**
## About the PR

Adds support for animated RSI states

## Technical details

For some reason it requires full rsi path

## Media

pog

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

### CODE REVIEW

**Review by lzk228 — CHANGES_REQUESTED** [review]
what about add `sprite: [path]` to `JobIcon` and remove it from all other

**Comment by Morb0** [comment]
> what about add `sprite: [path]` to `JobIcon` and remove it from all other

Nope, seems it require full sprite defenition
`Reading Content.Shared.StatusIcon.StatusIconPrototype(JobIconBotanist) threw the following exception: Robust.Shared.Serialization.InvalidMappingException: Expected sprite-node`

File: Content.Client/StatusIcon/StatusIconOverlay.cs [hunk 1]
```diff
@@ -72,9 +79,40 @@ protected override void Draw(in OverlayDrawArgs args)
...
+                            break;
+                        }
+
+                        texture ??= _sprite.Frame0(proto.Icon);
+                        break;
+                    case SpriteSpecifier.Texture spriteTexture:
+                        texture = spriteTexture.GetTexture(_resourceCache);
+                        break;
+                    default:
+                        throw new NotImplementedException();
```
Comment (metalgearsloth) [inline Content.Client/StatusIcon/StatusIconOverlay.cs:114]: Considering this code is shared with gastileoverlay it should be on a common system for handling it.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Comment by ps3moira** [comment]
it looks visually jarring and annoying to look at

**Comment by Morb0** [comment]
> it looks visually jarring and annoying to look at

sad

**Comment by AJCM-git** [comment]
@Morb0 you coming back to this?

**Pull request closed without merging**